task1.rb

```diff
@@ -0,0 +1,115 @@
+if (Gem.win_platform?)                                                # Для работы кириллицы
+  Encoding.default_external = Encoding.find(Encoding.locale_charmap)
+  Encoding.default_internal = __ENCODING__
+
+  [STDIN, STDOUT].each do |io|
+    io.set_encoding(Encoding.default_external, Encoding.default_internal)
+  end
+end
+
+class Station
+  attr_reader :name
+  def initialize(name)       #Инизиализация конструктора
+    @name=name
+    @trains=[]               #Объявление массива поездов
+  end
+  def add(train)             #Передаем объект класса Train
+    @trains << train         #Добавляем объект train в массив
+  end
+  def delete(train)                             #Удалить поезд, где в качестве аргумаента передается объект train
+    @trains.delete(train)                       #Альтернативный вариант удаления по индексу -- @trains.delete_at(train)
+  end
+  def show_trains                               #Выводим номера поездов
+    @trains.each { |train| puts train.number }  #Возможна и другая реализация, когда выводятся ID объектов -- attr_reader :trains -- пишется в начале класса
+  end
+  def show_f_trains
+    puts @trains.reject { |train| train.type=='freight'  }    #Вывод ID грузовых поездов
+    @trains.count  { |train| train.type=='freight'  }         #кол-во гр
+  end
+  def show_p_trains
+    puts @trains.reject { |train| train.type=='passenger'  }  #Вывод ID пассажирских поездов
+    @trains.count  { |train| train.type=='passenger'  }       #кол-во пс
+  end
+end
+
+
+class Route
+  attr_reader :arrstation
+  def initialize(beginning,final) #Передаем начальную и конечную станцию, то есть обекты Station
+    @arrstation=[]
```

Comment:
```suggestion
    @stations = []
```
чаще всего указание в имени переменной класса используемой коллекции излишне